### ISSUE TITLE
CDC cloud: changefeed sync latency 1+ hour after restart a CDC node

### ISSUE THREAD

**Issue by Tammyxia** [pingcap/tiflow#2551]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- There are 8 CDC servers, each has vcpu36c mem72g , 1 changefeed, 6w tables.
-  Add workload:  ./bin/go-ycsb run mysql -P workloads/betting -p operationcount=50000000 -p mysql.host=*.*.*.* -p mysql.port=* --threads 200 -p dbnameprefix=test_betting_ -p databaseproportions=1.0 -p unitnameprefix=unit1_ -p unitscount=100 -p tablecount=10000
- Check the changefeed latency is 2-3 minutes.
- Restart a VM which has a capture.

2. What did you expect to see?

- Changefeed works well, checkpoint latency less than 10m.

3. What did you see instead?

- Changefeed latency always keeps more than 1 hour after 3 hours.
![image](https://user-images.githubusercontent.com/78345569/129662083-d04e763b-3d79-4fc0-a5d3-25aaec8f5178.png)

- Upstrem tidb:
![image](https://user-images.githubusercontent.com/78345569/129662437-b3dd6459-2834-43aa-a377-5a6fed5237c9.png)

- Downstream is also a tidb cluster, its QPS much less than upstream tidb:
![image](https://user-images.githubusercontent.com/78345569/129662170-dc60313c-86f1-4b61-92b9-ec4f2257d0b3.png)

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        4.0.14
        ```

    - TiCDC version (execute `cdc version`):

        ```
        Release Version: v4.0.14-42-gc24e30f
        Git Commit Hash: c24e30f1a26b3de97443254b6c178acb613c2263
        Git Branch: HEAD
        UTC Build Time: 2021-08-11 06:03:32
        Go Version: go version go1.16.4 linux/amd64
        Failpoint Build: false
        ```

**Comment by Tammyxia** [comment]
#2229

**Comment by overvenus** [comment]
We have confirmed this issue is fixed in release-5.4.
